scipy/interpolate/interpolate.py

```diff
@@ -702,22 +703,20 @@ def _ensure_c_contiguous(self):
         if not self.c.flags.c_contiguous:
             self.c = self.c.copy()
 
-    def extend(self, c, x, right=True):
+    def extend(self, c, x):
```

Comment:
Removing keyword is backward compat break. Maybe retain it, but change the default value to `None` and emit a DeprecationWarning if the user gives a value different from `None`.